javalib/src/main/scala/java/security/MessageDigest.scala

```diff
@@ -27,3 +40,44 @@ private class DummyMessageDigest(algorithm: String)
                                       offset: Int,
                                       len: Int): Unit = ()
 }
+
+@link("crypto")
+@extern
+private object crypto {
+  def MD5_Init(c: Ptr[Byte]): CInt                                = extern
+  def MD5_Update(c: Ptr[Byte], data: Ptr[Byte], len: CSize): CInt = extern
+  def MD5_Final(md: CString, c: Ptr[Byte]): CInt                  = extern
+}
+
+private final class Md5MessageDigest(algorithm: String)
+    extends MessageDigest(algorithm) {
+  import crypto._
+
+  // Array with length equals to sizeof(MD5_CTX)
+  private val c = new Array[Byte](92).asInstanceOf[ByteArray].at(0)
```

Comment:
Are you sure that all `libssl` implementation has this size?

Reply:
I have to check, I'm not sure.
I first tried to fetch it with a C file but that creates a dependency to Openssl (all the files in `/resources/scala-native` regardless if you use them or not), so I checked this number on my Mac OS machine with OpenSSL.